src/helpers/tagHelpers.js

```diff
@@ -29,5 +29,26 @@ class TagHelpers {
       return error.message;
     }
   }
+
+  /**
+   * @description - This method is responsible for retrieving all the top tag names
+   * @static
+   * @param {object} topTags - Array of the top tag IDs
+   * @returns {object} - an array of tags objects with name property
+   * @memberof TagHelpers
+   */
+  static async findTags(topTags) {
+    try {
+      const allTags = await Promise.all(topTags.map(tag => Tag.findOne({
```

Comment:
Good forward thinking here using `Promise.all`, however, don't you think hitting the database 7 times (if an array of 7 trending tag ids is passed in) is a wee bit too much? What if an array of 50 tag ids was passed? We'll query the database 50 times?

_Here's a proposition:_

What if we just get all the tags in the database, then map over that recursively (or using a forEach) filtering that to get the names? That way, we hit the database **just once**, then use JS to massage the data which comes back.

What do you think?

Reply:
hmm, @akhilome nice observation there, you're right, that doesn't seem efficient enough, I'll rethink It using your proposition and make changes where necessary, thanks. 👍

Reply:
Hi @akhilome I've implemented your changes and I learnt a lot while doing so, Thanks for making me embark on such a task